config.py

```diff
@@ -89,6 +91,34 @@ def known_folder_path(guid: uuid.UUID) -> Optional[str]:
 _T = TypeVar('_T')
 
 
+###########################################################################
+def git_shorthash_from_head() -> str:
+    """
+    Determine short hash for current git HEAD.
+
+    :return: str - None if we couldn't determine the short hash.
+    """
+    shorthash: str = None  # type: ignore
```

Comment:
If we try Optional[str] throughout here then the file write() call in setup.py complains because it wants only `str`.  So hide the issue here.